### PR TITLE
Add `scheme_autogeneration_mode` with support for `none`, `auto`, and `all`

### PR DESCRIPTION
**Pull request by cgrindel**
Related to #573.

### CODE REVIEW

File: tools/generator/src/DTO/SchemeAutogenerationMode.swift [hunk 1]
```diff
@@ -0,0 +1,5 @@
+enum SchemeAutogenerationMode: String, Decodable {
```
Comment (brentleyjones) [inline tools/generator/src/DTO/SchemeAutogenerationMode.swift:1]: Is `top_level` coming later?

File: xcodeproj/internal/xcodeproj.bzl [hunk 2]
```diff
@@ -99,7 +99,8 @@ def _write_json_spec(*, ctx, project_name, configuration, inputs, infos):
 "label":"{label}",\
 "name":"{name}",\
 "target_merges":{target_merges},\
-"targets":{targets}\
+"targets":{targets},\
+"scheme_autogeneration_mode":"{scheme_autogeneration_mode}"\
```
Comment (brentleyjones) [inline xcodeproj/internal/xcodeproj.bzl:103]: Can this be moved up to be alphabetical?

File: tools/generator/src/DTO/SchemeAutogenerationMode.swift (hunk 1 above)
Comment (cgrindel) [reply tools/generator/src/DTO/SchemeAutogenerationMode.swift:1]: Yes. I created a separate TODO. Adding it was more than just plumbing the value through and adding a simple `guard`. I figured that it deserved its own PR.

File: xcodeproj/internal/xcodeproj.bzl (hunk 2 above)
Comment (cgrindel) [reply xcodeproj/internal/xcodeproj.bzl:103]: Yes. Apparently, I am pretty sleepy this afternoon. I thought that it was alphabetical. 😳

File: test/fixtures/tvos_app/bwx_spec.json [hunk 3]
```diff
@@ -419,5 +419,6 @@
             "swiftmodules": [],
             "test_host": null
         }
-    ]
+    ],
+    "scheme_autogeneration_mode": "auto"
```
Comment (brentleyjones) [inline test/fixtures/tvos_app/bwx_spec.json:423]: Hmm. Why is it still showing at the bottom here? Do the fixtures need regenerating?

Comment (cgrindel) [reply test/fixtures/tvos_app/bwx_spec.json]: Sorry. I forgot to regenerate the fixtures after updating.

**Review by brentleyjones — APPROVED** [review]

**Pull request merged**